libc/newhdrgen/class_implementation/classes/function.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+#
+# ====-- Function class for libc function headers -------------*- python -*--==#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ==-------------------------------------------------------------------------==#
+
+
+class Function:
+    def __init__(self, return_type, name, arguments, guard=None, attributes=None):
+        self.return_type = return_type
+        self.name = name
+        self.arguments = [arg if isinstance(arg, str) else arg['type'] for arg in arguments]
```

Comment:
arguments is an array of strings or dictionaries? Can we just make it always one (or the other)?